Formula/curl.rb

```diff
@@ -67,6 +77,21 @@ def install
     args << (build.with?("gssapi") ? "--with-gssapi" : "--without-gssapi")
     args << (build.with?("rtmpdump") ? "--with-librtmp" : "--without-librtmp")
 
+    if build.with? "libidn2"
+      resource("libidn2").stage do
+        system "./configure", "--disable-dependency-tracking",
+                              "--disable-silent-rules",
+                              "--prefix=#{libexec}/vendor/libidn2",
```

Comment:
Didn't test, to be honest. I just automatically lean towards shared library use on macOS given that seems to be Apple's preference more often than not.